### PR TITLE
Implement updates from internal review

### PR DESCRIPTION
**Pull request by spengrah**
- changes the name to HatsAccount
- cleans up comments, natspec, and dependencies
- more sensible rejection threshold
- more efficient valid vote counting

### CODE REVIEW

File: src/HatsAccountMofN.sol [hunk 1]
```diff
@@ -496,16 +512,12 @@ contract HatsWalletMofN is HatsWalletBase {
           }
         }
```
Comment (gershido) [inline src/HatsAccountMofN.sol]: Assuming that the client should provide only valid votes, we can do this:
```typescript
if (!votes[_proposalId][currentVoter] == _vote || !_isValidSigner(currentVoter) {
   revert InsufficientValidVotes();
}
```
And then remove `count` and the ```if (count < _threshold) revert InsufficientValidVotes();``` condition at the end.

**Review by gershido — COMMENTED** [review]
Regarding the following function:  
```solidity
  function isExecutableNow(bytes32 _proposalId, address[] calldata _voters) external view returns (bool) {
    _checkExecutableNow(_proposalId, _voters);
    return true;
  }
```
`_checkExecutableNow` reverts if not executable, returns `true` if yes. A slightly different design here can be that `_checkExecutableNow` won't have a return value at all. Then the view function `isExecutableNow` can have the actual logic and return true/false, and the internal`_checkExecutableNow` will use it and revert or not. Same also with `isRejectableNow`

File: src/HatsAccountMofN.sol (hunk 1 above)
Comment (gershido) [reply src/HatsAccountMofN.sol]: Above comment is regarding [`_checkValidVotes`](https://github.com/Hats-Protocol/hats-account/blob/fa3dc812ef88355658d17140d561e73af5f12c95/src/HatsAccountMofN.sol#L509-L513)

**Comment by gershido** [comment]
Wonder if we want to add the tx parameters to the `TxExecuted` event:
```solidity
  function execute(address _to, uint256 _value, bytes calldata _data, uint8 _operation)
    external
    payable
    returns (bytes memory result)
  {
    if (!_isValidSigner(msg.sender)) revert InvalidSigner();


    // increment the state var
    _beforeExecute();


    // execute the call, routing delegatecalls through the sandbox, and bubble up the result
    result = LibHatsAccount._execute(_to, _value, _data, _operation);


    // log the executor
    emit TxExecuted(msg.sender);
  }
  ```

**Comment by spengrah** [comment]
> Wonder if we want to add the tx parameters to the `TxExecuted` event:

@gershido my thinking in leaving them out was that we could get them from the tx args. But happy to include them in the event if that would improve things on the subgraph side

**Comment by spengrah** [comment]
added improvements based on discussion with @gershido:
- even more efficient valid vote counting
- simplify/consolidate logic for `_checkExecutableNow` and `isExecutableNow` and Rejectable equivalents

may add tx params to the `HatsAccount1ofN.TxExecuted` event pending @gershido's subgraph research

**Comment by gershido** [comment]
Lgtm!

Regarding the `HatsAccount1ofN.TxExecuted` event, it won't be a problem to index the call info with the current implementation.

**Review by gershido — APPROVED** [review]

**Pull request merged**